### DISCUSSION TITLE
Security Question

### DISCUSSION THREAD

**Discussion by DosParkers** [d2phap/ImageGlass#6] [General] [answered]
Does ImageGlass save any data outside of it's install folder to the disc? Like thumbnails and/or caches.

**Comment by d2phap** [comment, marked as answer]
Hello,

There are 2 kinds of data saved:

**1. ImageGlass Configurations:**
All configurations will be saved in registry HKEY_CURRENT_USER\Software\PhapSoftware\ImageGlass\
You can find it here: https://github.com/d2phap/ImageGlass/blob/master/Source/Extensions/ImageGlass.Services/Configuration/GlobalSetting.cs#L372

**2. ImageGlass Plugins:**
Language packs and extensions are saved in install folder. However, due to security of Windows OS, ImageGlass theme packs are stored in %AppData%\ImageGlass\
You can check it here:
https://github.com/d2phap/ImageGlass/blob/master/Source/Commands/igcmd/frmInstallTheme.cs#L64

In the future, I will consider to storing other data, such as extensions, language packs in the same place of theme pack's
